### ISSUE TITLE
useOverlayPosition: strange behaviour of shouldUpdatePosition

### ISSUE THREAD

**Issue by LosYear** [adobe/react-spectrum#4477]
Hey!

I think that the naming of `shouldUpdatePosition` is a little bit confusing. As I imagine, it should calculate position once and then ignore all resizes, scrolls, etc on subsequent updates

But according to line:
https://github.com/adobe/react-spectrum/blob/273e828cd17c3d229e8ab10ce62bd0a6f0f43963/packages/%40react-aria/overlays/src/useOverlayPosition.ts#L128

If shouldUpdatePosition is false, it will never calculate position even for initial render. I believe it should work like `eventListeners` modifier of Popper.js (https://popper.js.org/docs/v2/modifiers/event-listeners/)

This behaviour leads bug/behaviour like this:
https://codesandbox.io/s/great-mendel-l0c8ye?file=/src/App.js
wrong initial positioning with shouldUpdatePosition=false, since useOverlayPosition does nothing useful in this case

If you're agree with it and it won't break anything, let me submit a PR :)

**Comment by LFDanLu** [comment]
I agree that `shouldUpdatePosition` is pretty confusing, and I don't think we even make use of it anymore. Do you have a particular use case in mind for this updated behavior? I like the idea but I just want to make sure we have a clear intent with the change. 

Off the top of my head, there may be some complexity when determining the proper initial position to freeze it at since our popovers take 2+ renders to settle on a initial position (one render for the popover positioning  and one for arrow positioning if I remember correctly).

**Comment by LosYear** [comment]
I agree that it looks confusing.

Honestly, seems like `shouldFlip` covers all our cases when popup should not be repositioned after opening.

I see, it really takes min. 2 renders to position it correctly. But does it takes exactly 2? Can we rely on this?

**Comment by LFDanLu** [comment]
I'd have to dig through all the existing cases, but I don't think we can rely on the assumption that it takes exactly 2 renders. Quite frankly, the team has been discussing using/try floatingui/popper to handle overlay positioning instead of our current implementation due to these gotchas that have been popping up over time...

**Comment by LosYear** [comment]
That sounds like a great idea since popper seems to be a de facto standard for overlay positioning

But for now, may be we should drop the `shouldUpdatePosition`?

**Comment by LFDanLu** [comment]
Thinking out loud, that would be a bit tricky since that would be a breaking change if someone was relying on the current behavior to essentially turn off our calculated positioning and control positioning themselves.

**Comment by agusterodin** [comment]
I had to use this hacky workaround to get an initial position but make it so that after that popover no longer moves around. The idea is that I don't want the calendar next and previous buttons to shift under the user's mouse for months that have an extra row of days.

Here is the workaround:

<img width="1008" alt="image" src="https://github.com/adobe/react-spectrum/assets/10248395/99117a86-6d88-4872-b3a4-de23d129d879">

And here is the result (notice popover stays perfectly still despite growing):

https://github.com/adobe/react-spectrum/assets/10248395/08d50bd8-8cfb-4c31-8ce7-ee249bc56a98

The popover arrow now no longer works in recent versions of React ARIA

https://github.com/adobe/react-spectrum/assets/10248395/1c1ff064-7d84-49a4-beac-e3ad7d94d82c

Do you have a suggested way of fixing the broken popover arrow? Do you have a suggested alternative to make sure position doesn't change as popover grows/shrinks? Please don't get rid of shouldUpdatePosition otherwise :(

**Comment by agusterodin** [comment]
Fixed the issue shown above in latest version of React ARIA with this even more disgusting hack.

Please don't get rid of shouldUpdatePosition though! It has a legitimate use case.

<img width="872" alt="image" src="https://github.com/adobe/react-spectrum/assets/10248395/9245d669-82ad-45fe-9fa0-e310b838b9a2">

**Comment by LFDanLu** [comment]
Happy to see that you were able to get a workaround, and thanks for letting us know about your use case! I don't think we will be removing shouldUpdatePosition, what we've been considering so far is reverting the change where we are applying the [styles directly on the popover DOM node](https://github.com/adobe/react-spectrum/blob/273e828cd17c3d229e8ab10ce62bd0a6f0f43963/packages/%40react-aria/overlays/src/useOverlayPosition.ts#L147-L150) which may be the cause of your previously noted break in behavior. 
That change been causing some other issues as noted [here](https://github.com/adobe/react-spectrum/issues/1924#issuecomment-1524127447), with a potential fix to investigate.
